csharp/autobuilder/Semmle.Autobuild.Shared/DiagnosticClassifier.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Semmle.Util;
+
+namespace Semmle.Autobuild.Shared
+{
+    /// <summary>
+    /// Direct results result from the successful application of a <see cref="DiagnosticRule" />,
+    /// which can later be converted to a corresponding <see cref="DiagnosticMessage" />.
+    /// </summary>
+    public interface IDiagnosticsResult
```

Comment:
If there's no strong reason to keep these classes and interfaces in this file, I'd go with separate files for the declarations.